src/nvim/api/vim.c

```diff
@@ -101,11 +102,16 @@ String nvim_exec(String src, Boolean output, Error *err)
   if (output) {
     msg_silent++;
   }
+
+  const sctx_T save_current_sctx = api_set_sctx(channel_id);
```

Comment:
starting to see a need for a common preamble/teardown wrapper that every API function should have.

Reply:
Is this a comment ? Or a change request I'm confused .

Reply:
either? should probably wait for a followup PR though, to avoid risk here.